packages/react-theme-provider/src/components/ThemeProvider/useThemeStyleTag.ts

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { useIsomorphicLayoutEffect, useId } from '@fluentui/react-utilities';
+import { themeToCSSVariables } from '@fluentui/react-theme';
+import { ThemeProviderState } from './ThemeProvider.types';
+
+/**
+ * Writes a theme as css variables in a style tag on the provided targetDocument as a rule applied to a CSS class
+ *
+ * @param theme Theme values that are written as css variables in style rule
+ * @returns CSS class to apply the rule
+ */
+export const useThemeStyleTag = (
+  theme: ThemeProviderState['theme'],
+  targetDocument: ThemeProviderState['targetDocument'],
+) => {
+  const styleTagId = useId('theme-provider');
+  const styleTag = getOrCreateStyleTag(styleTagId, targetDocument);
```

Comment:
This will handle `document` updates only partially as `style` node will not be removed from a previous document.

I made an example on CodeSandbox that will cleanup it properly: https://codesandbox.io/s/stupefied-dhawan-resst?file=/src/App.tsx


It will also remove a need in `useIsomorphicLayoutEffect()` below.